homeassistant/components/todoist/calendar.py

```diff
@@ -594,13 +594,14 @@ async def async_get_events(self, hass, start_date, end_date):
             )
             if not due_date:
                 continue
-            midnight = dt.as_utc(
-                dt.parse_datetime(
-                    due_date.strftime("%Y-%m-%d")
-                    + "T00:00:00"
-                    + self._api.state["user"]["tz_info"]["gmt_string"]
-                )
+            gmt_string = self._api.state["user"]["tz_info"]["gmt_string"]
+            local_midnight = dt.parse_datetime(
+                f'{due_date.strftime("%Y-%m-%d")}T00:00:00{gmt_string}'
             )
```

Comment:
I think this can also work since strftime will only evaluate the % and can accept an f-string itself:

```suggestion
            local_midnight = dt.parse_datetime(
                due_date.strftime(f"%Y-%m-%dT00:00:00{gmt_string}")
            )
```